### PR TITLE
Domains based LDAP and AD Fullname and Email not auto-populated

### PR DESCRIPTION
**Pull request by dk-dksoft**
Patch to resolve issue: https://github.com/Cacti/cacti/issues/4768

### CODE REVIEW

**Review by TheWitness — CHANGES_REQUESTED** [review]
Okay, here are my comments and there are definately changes require:

1. In auth_login.php on line 549, you mis-spelled the function name.
2. In the function domain_ldap_search_cn() there are a few errors. First, on line 663, you are not using tabs for indentation.  Looks like spaces at the beginning.
3. On that very same line, our PSR requires that all if/then/else be enclosed in curly braces, so "if (condition) { blah(); } else { blah(); }
4. Delete line 659.  There are PSR versions that require it, but not Cacti.
5. You have a few blank lines between 690 and 693 and I see more soft/hard tab issues.

I would not be surprised if you grabbed this function strait from Cacti's lib/ldap.php, and again, we break our own rules in the whole set of if blocks around that whole function.

I had fixed this in a project in the Winter of 2021, but we backed the whole thing out.  If you could make these changes, then we can merge.  Might not be till after the 1.2.21 release though.

**Comment by TheWitness** [comment]
@dk-dksoft, are you planning on fixing your commit?

**Comment by dk-dksoft** [comment]
Hello,
Thanks for comments!  Yes, i will make changes in a few days!

вс, 15 мая 2022 г., 02:41 TheWitness ***@***.***>:

File: auth_login.php [hunk 1]
```diff
@@ -546,8 +546,7 @@ function domains_login_process() {
 							array(get_nfilter_request_var('realm')-1000));
 
 						if ($cn_full_name != '' || $cn_email != '') {
-							$ldap_cn_search_response = cacti_ldap_search_cn($username, array($cn_full_name,$cn_email) );
-
+							$ldap_cn_search_response = domain_ldappp_search_cn($username, array($cn_full_name,$cn_email),get_nfilter_request_var('realm')-1000);
```
[the remark this answers is not in the document]
Comment (dk-dksoft) [reply auth_login.php]: sorry for that , i have copied it from my working enviroment but changed name in web . Now All is ok .

**Comment by TheWitness** [comment]
We just need a CHANGELOG entry now.  Thanks!

**Comment by TheWitness** [comment]
Closes #4819 - Domains based LDAP and AD Fullname and Email not auto-populated

**Pull request merged**